### ISSUE TITLE
0x5 Obtain Android ADB Access (Optional, Subject to Change)  这一步刷完以后，一直显示Prepare for updating...

### ISSUE THREAD

**Issue by wangyongxina** [HappyZ/dpt-tools#74]
然后我reset了，现在只有黄灯在闪？咋整？

**Comment by yefeiyu** [comment]
My understanding, theoretically speaking, the `fw` mode is very safe...

**Comment by wangyongxina** [comment]
> My understanding, theoretically speaking, the `fw` mode is very safe...

I've waited about twenty minutes, but dpt rp1 always show Prepare for updating, 
and finaly I try reset, and wait for reboot. but it cannot boot into normal system

**Comment by HappyZ** [comment]
in any cases pressing reset button should be avoided.. but I guess you have patched the update script and you have three bars appearing on the top left corner?

if so, you should be able to hold HOME button to get into diagnosis mode, or hold POWER button to remove the PKG. 
Plz refer to [this wiki](https://github.com/HappyZ/dpt-tools/wiki/Unbrick-and-or-Skip-Uploaded-PKG)

**Comment by wangyongxina** [comment]
Thank HappyZ, I've fixed brick through taobao guy, Now dpt rp1 can boot into normal system. I try to continue Obtaining Android ADB Access step, This time fw is work ok, but dpt rp1 cannot get Android ADB Access, I've tried many times. and finally, I try to recover system to normal using [this link](https://github.com/HappyZ/dpt-tools/issues/31), also failed. 
then I tried to login diagnosis mode using dpt-tools, it show login failed, the password is wrong.
It seems the root password is *reset* after taobao guys fixed my devices.
I even tried to using shankerzhiwu_changepwd.pkg to reset password, but failed.
Any suggestion?
thanks

**Comment by HappyZ** [comment]
yes they probably clears the entire system for you.. huh that's interesting. 

You mean you could still enter diagnosis mode (seeing the block), but just could not log in? Do you still have the three bars on the top left? Just to make sure.

`shankerzhiwu_changepwd.pkg` should say `failed` on screen but the password changing should be finished. 

However, if taobao guy changed your update script and fixed the updating script bug, then the hack will no longer work.

**Comment by wangyongxina** [comment]
Hi HappyZ, 
I try to use shankerzhiwu_changepwd.pkg to reset password then enter diagnosis mode, but failed at login. It seems my device's fw function is broken. 
I event try to modify the password to empty, but cannot login.

I can only see block when I try to enter diagnosis mode.

when boot up, I press physical home button, but I cannot see three bars on the top left, only see lights  blink, and then enter diagnosis mode directly. I even tried press physical home button when I see welcome... at booting.

I ask the taobao guy, how he fixed my device, but he doesn't tell me.
Is  there any way I can recover my system to original?

thanks

**Comment by HappyZ** [comment]
I see, that should be ok. Yours just get reset to the very original version, I guess. I have just updated the single PKG that hacks all. Try it and let me know if the screen prints anything.

Note: I have not finished the Android part of it, so stay tuned after step 0x2 in the guide.

**Comment by HappyZ** [comment]
Also I don't think `fw` would be broken. It is a system function. It shall work anytime as long as you can boot into the system.

**Comment by wangyongxina** [comment]
HI  HappyZ, After using fw to apply fw_updater_packer_by_shankerzhiwu\pkg_example\hack_basics\fw.pkg,  My device only show Prepare updating, Welcome..., No other information print. I try to enter diagnosis mode, But it failed at Login.

**Comment by HappyZ** [comment]
i can only guess your updater script has been "fixed" so our hack doesn't work any more.. hmmm.. probably wait till someone was exploiting web interface.. or contact the seller and ask what he's changed

**Comment by HappyZ** [comment]
Closed as there are no more progressive discussions. If you got new questions, please submit a new issue.

To conclude here, if you cannot use `fw` to flash a hacking PKG and change the password, then highly likely your updater script is fixed and bug free. Therefore our attack method fails.
